### DISCUSSION TITLE
Make this window editable

### DISCUSSION THREAD

**Discussion by MarvinKlein1508** [dbeaver/dbeaver#19621] [Q&A] [answered]
![grafik](https://user-images.githubusercontent.com/32510006/228486327-bbfb5bf4-1deb-4c79-ac63-4445bdcf7ebe.png)

I want to edit this SQL statement because I don't need to see all columns all the time. It would make searching for data much easier.

**Comment by LonwoLonwo** [comment]
Hello @MarvinKlein1508 

Could you provide any examples of what you want?
For now, it needs to be clarified. This window just shows the current query used for the data reading in the Data Editor.

**Comment by MarvinKlein1508** [comment]
@LonwoLonwo sure. I want to manipulate this query. Let's say all I care about is the ID and the username and I don't want to see all the other columns. So I would change it to select id, username from.

Right now to achive this I'll need to write the query on my own. It would be quite nice if the table itself offers this functionatlity somehow.

**Comment by MarvinKlein1508** [comment]
When doing a new script file, it's hard to tell what this script file is about because it just displays new script in the tabs. So I think it would be a great addition to the table itself. This way I know the table I'm dealing with.

**Comment by uslss** [comment, marked as answer]
@MarvinKlein1508 did you try to use [Advanced filters](https://github.com/dbeaver/dbeaver/wiki/Data-Filters#advanced-filters-configuration)? here you can change column visibility

**Comment by MarvinKlein1508** [comment]
@uslss ah thanks! This is exactly what I was looking for. 

Maybe you can help me out with this question as well. Is there any way to rename those tabs without saving the file?
![grafik](https://user-images.githubusercontent.com/32510006/230060432-f2a93c43-76f0-4ae5-b09e-3262b8f5a727.png)

I'm doing multiple queries to merge them all together into one statement and naming those tabs would help me out enormously to identify what I was working with here.

**Comment by E1izabeth** [comment]
As far as I know, there is no way to rename SQL Script tab without renaming a saved file.
